### PR TITLE
nd_interface_subinterface_unmanaged

### PR DESCRIPTION
**Pull request by allenrobel**
## Related Issue(s)

References CiscoDevNet/ansible-nd#295 (the 207-status:failed silent-success bug in `response_handler_nd.py` is worked around in this module via `_raise_on_multi_status_failures`; remove the workaround when #295 lands).

**Note:** This PR is stacked on #296 (`nd_interface_subinterface_managed`). Please review #296 first; this PR will be retargeted to `develop` and rebased once #296 merges.

- CiscoDevNet/ansible-nd#325

### Deferred follow-ups (tracked)

- #348 — interface orchestrator CRUD + policy-type serializer dedup (cross-cutting; this module duplicates the managed sibling)
- #349 — query_all per-switch fan-out / no fabric-wide interface endpoint (cross-cutting)

## Proposed Changes

- Add the `nd_interface_subinterface_unmanaged` module to manage L3 subinterfaces in monitor mode (unmanaged variant, `policyType: monitorSubinterface`) on Cisco Nexus Dashboard 4.2.
- Add `SubinterfaceUnmanagedInterfaceModel` and `SubinterfaceUnmanagedInterfaceOrchestrator`, mirroring the managed variant's structure but with the policy body reduced to its `policyType` discriminator only.
- All scaffolding fields (`interface_type`, `mode`, `network_os_type`, `policy_type`) are hardcoded via `Literal[...] + Field(frozen=True)` and hidden from the user-facing argument spec. The user supplies only `switch_ip` and `interface_name` per item.
- `query_all` filters on `policyType in {monitorSubinterface}` via a dedicated `SubinterfaceUnmanagedPolicyTypeEnum`, so managed and unmanaged subinterfaces are cleanly partitioned across the two modules.
- Defensive `extra="ignore"` on the policy model absorbs any read-only fields ND echoes under `policy` on GET (e.g. the schema's `userDefined` discriminator branch siblings).
- Orchestrator unit tests at 98% coverage (23 tests, including the `_raise_on_multi_status_failures` matrix and a positive assertion that `query_all` filters out the managed `subinterface` policyType).
- Integration test target with sub-id range `.20`-`.29` to coexist with the managed target's `.2`-`.5` range. Phases: merged (single, multi, Port-channel parent, idempotency), replaced (no-op on existing + fall-back-to-create on missing), overridden (fabric-wide reduce), deleted.

## Test Notes

- Integration tests pass on ND 4.2.1 lab SITE1 against routed-mode parents (`Ethernet1/3` and `Port-channel10` on `S1_TOR1`). Both managed and unmanaged targets were run successfully back-to-back to confirm the sub-id ranges do not collide.
- 23 unit tests pass (`python -m pytest tests/unit/module_utils/orchestrators/test_subinterface_unmanaged_interface.py`).
- `ansible-test sanity --docker` clean on the touched files.
- Two carried-over ND 4.2.1 wire workarounds:
  - Parent must be in routed (L3) mode — same as managed; surfaced via `_raise_on_multi_status_failures` and documented in `notes:`.
  - `interface_name` echoed lowercased on GET — same `normalize_interface_name` field validator pattern as the managed variant.
- The `routing_tag` int→str workaround does NOT apply here (no policy fields with that name in the unmanaged body).

## Cisco Nexus Dashboard Version

4.2.1

## Related ND API Resource Category

* [ ] analyze
* [ ] infra
* [x] manage
* [ ] onemanage
* [ ] other

## Checklist

* [x] Latest commit is rebased from develop with merge conflicts resolved
* [x] New or updates to documentation has been made accordingly
* [ ] Assigned the proper reviewers

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/modules/nd_interface_subinterface_unmanaged.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        - This is resolved to the switch serial number (switchId) internally.
+        type: str
+        required: true
+      interface_name:
+        description:
+        - The full subinterface name, including the dot-separated sub-id (e.g. C(Ethernet1/3.20), C(Port-channel10.20)).
+        - The parent kind is inferred from the prefix; only Ethernet and Port-channel parents are supported.
+        type: str
+        required: true
+  deploy:
```
Comment (mikewiebe) [inline plugins/modules/nd_interface_subinterface_unmanaged.py]: ### Medium: Replace top-level `deploy` with `config_actions`

**Issue**

The module exposes a single top-level `deploy` boolean instead of the standard `config_actions.save`, `config_actions.deploy`, and `config_actions.type` control shape.

**Evidence**

- [`plugins/modules/nd_interface_subinterface_unmanaged.py` lines 53-77](https://github.com/CiscoDevNet/ansible-nd/pull/297/files#diff-3d6dd4799540245b3fe111a0d31cd31ee2c2b6bde61fd0f342650614c1b95ed1R53) documents `deploy` as a standalone option and lists states without the standard `config_actions` option.
- [`plugins/modules/nd_interface_subinterface_unmanaged.py` line 181](https://github.com/CiscoDevNet/ansible-nd/pull/297/files#diff-3d6dd4799540245b3fe111a0d31cd31ee2c2b6bde61fd0f342650614c1b95ed1R181) adds `deploy=dict(type="bool", default=True)` directly in the wrapper argspec.
- The bundled OpenAPI spec exposes `POST /fabrics/{fabricName}/interfaceActions/deploy`, so deploy behavior is an explicit API capability rather than a private implementation detail.
- For a committed implementation example, [`plugins/modules/nd_manage_switches.py` lines 47-72](https://github.com/CiscoDevNet/ansible-nd/blob/develop/plugins/modules/nd_manage_switches.py#L47-L72) documents the standard `config_actions` option with `save`, `deploy`, and `type`.

**Existing PR overlap**

No matching existing PR comment found.

**Impact**

Keeping a one-off `deploy` option makes the user interface drift from the newer module contract and gives users no consistent place to control save/deploy behavior across modules.

**Suggested fix**

Replace `deploy` with `config_actions` using the same `save`, `deploy`, and `type` shape used by `nd_manage_switches`. Add at least one example that shows `config_actions` and update tests to verify that `config_actions.deploy: false` suppresses the deploy call.

File: plugins/modules/nd_interface_subinterface_unmanaged.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    description:
+    - Whether to deploy interface changes after mutations are complete.
+    - When V(true), all queued interface changes are deployed in a single bulk API call at the end of module execution
+      via the C(interfaceActions/deploy) API. Only the subinterfaces modified by this task are deployed.
+    - When V(false), changes are staged but not deployed. Use a separate deploy module or task to deploy later.
+    - Setting O(deploy=false) is useful when batching changes across multiple interface tasks before a single deploy.
+    type: bool
+    default: true
+  state:
```
Comment (mikewiebe) [inline plugins/modules/nd_interface_subinterface_unmanaged.py:67]: ### Low: Track `gathered` support against issue #340 (non-blocking)

**Issue**

Tracking note only, not a blocking request for this PR: the module and model expose only `merged`, `replaced`, `overridden`, and `deleted`. Adding `gathered` for modules that use the `NDStateMachine` plus orchestrator pattern is already being tracked in CiscoDevNet/ansible-nd#340.

**Evidence**

- [`plugins/modules/nd_interface_subinterface_unmanaged.py` lines 62-77](https://github.com/CiscoDevNet/ansible-nd/pull/297/files#diff-3d6dd4799540245b3fe111a0d31cd31ee2c2b6bde61fd0f342650614c1b95ed1R62) documents state choices without `gathered`.
- [`plugins/module_utils/models/interfaces/subinterface_unmanaged_interface.py` lines 202-206](https://github.com/CiscoDevNet/ansible-nd/pull/297/files#diff-55d3f823f6d32df3fe77398b7f11845314e4bdaa1d1c75ea73730fb111adbac4R202) returns the same state choices from the model argspec.

**Existing PR overlap**

No matching existing PR comment found.

**Impact**

Leaving `gathered` out for now is acceptable while issue #340 tracks the collection-wide addition. Calling it out here just keeps this module on the list so the read-only workflow can be added consistently across ND state-machine backed modules.

**Suggested fix**

No immediate PR-blocking change is required. When CiscoDevNet/ansible-nd#340 is implemented, add `gathered` support with documented filter semantics, at least one `gathered` example, and a meaningful test that verifies the read-only output and that no create/update/delete/deploy call is made.

---

File: plugins/module_utils/models/interfaces/subinterface_unmanaged_interface.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+    # --- Fields ---
+
+    switch_ip: str = Field(alias="switchIp")
+    interface_name: str = Field(alias="interfaceName")
+    interface_type: Literal["subInterface"] = Field(default="subInterface", alias="interfaceType", frozen=True)
+    config_data: SubinterfaceUnmanagedConfigDataModel = Field(default_factory=SubinterfaceUnmanagedConfigDataModel, alias="configData")
+    oper_data: SubinterfaceUnmanagedOperDataModel | None = Field(default=None, alias="operData")
+
+    @field_validator("interface_name", mode="before")
```
Comment (mikewiebe) [inline plugins/module_utils/models/interfaces/subinterface_unmanaged_interface.py:139]: ### Medium: Add focused model validation tests for subinterface name handling

**Issue**

The model adds subinterface name normalization and validation, but the PR does not include a focused model test file for those validator edge cases.

**Evidence**

- [`plugins/module_utils/models/interfaces/subinterface_unmanaged_interface.py` lines 138-172](https://github.com/CiscoDevNet/ansible-nd/pull/297/files#diff-55d3f823f6d32df3fe77398b7f11845314e4bdaa1d1c75ea73730fb111adbac4R138) validates dotted `Ethernet...` and `Port-channel...` names and normalizes parent casing.
- The PR adds `tests/unit/module_utils/orchestrators/test_subinterface_unmanaged_interface.py`, but no `tests/unit/module_utils/models/test_subinterface_unmanaged_interface.py` file.

**Existing PR overlap**

No matching existing PR comment found.

**Impact**

This validator is part of the runtime user contract and idempotency behavior. Without focused model tests, regressions in accepted prefixes, missing dot handling, invalid parent rejection, or response normalization can slip through while higher-level orchestrator tests still pass.

**Suggested fix**

Add model tests for valid Ethernet and Port-channel subinterface names, lowercase ND response normalization, missing subinterface suffix, invalid parent prefix, empty/non-string passthrough behavior, and payload/diff serialization of the fixed hidden fields.

File: plugins/modules/nd_interface_subinterface_unmanaged.py (hunk 1 above)
Comment (allenrobel) [reply plugins/modules/nd_interface_subinterface_unmanaged.py]: Addressed in ace59646.

Replaced the top-level `deploy` bool with a `config_actions` dict carrying a nested `deploy` suboption, mirroring the same migration on `nd_interface_ethernet_trunk_host` (ec82a6f2), `nd_interface_svi` (d5afacfb), and `nd_interface_subinterface_managed` (8806d87d):

- **DOCUMENTATION** — `deploy:` → `config_actions:` (`type: dict`) with `deploy` as a suboption (`type: bool`, `default: true`).
- **argspec** — `config_actions` dict with `options: {deploy: {type: bool, default: True}}`.
- **orchestrator bridge** — reads `config_actions.deploy` (default `True`) into `model_orchestrator.deploy`.
- **EXAMPLES** — the no-deploy/batching example now uses the nested `config_actions: {deploy: false}` form.

On `save`/`type`: intentionally omitted here (same as the other interface modules). Verified against the ND 4.2.1 OpenAPI spec that `interfaceActions/deploy` is interface-type agnostic (body is `{interfaces:[{interfaceName,switchId}]}`), has no separate save/recalculate step, and exposes no switch|global scope parameter — a global scope would also deploy unrelated pending fabric config, which the interface resource modules deliberately avoid for safety. Happy to add `save`/`type` if you'd prefer full parity with `nd_manage_switches`.

Verification: `validate-modules` sanity clean, `black` clean.

File: plugins/module_utils/models/interfaces/subinterface_unmanaged_interface.py (hunk 3 above)
Comment (allenrobel) [reply plugins/module_utils/models/interfaces/subinterface_unmanaged_interface.py:139]: Addressed in 01d2635a.

Added `tests/unit/module_utils/models/test_subinterface_unmanaged_interface.py` (27 cases) to sit alongside the existing orchestrator tests. It covers the validator edge cases and fixed-field serialization you called out:

- **interface_name normalizer** — valid Ethernet and Port-channel names; lowercase/uppercase/mixed-case ND-response normalization to canonical casing; padded (whitespace-stripped); missing-`.`-subid rejection; non-Ethernet/Port-channel parent rejection; empty-string and non-string (None) passthrough through the guard (then deferred to `str` type validation).
- **Hidden/fixed fields** — the `monitorSubinterface` policy discriminator default, the `networkOS`/`configData` default_factory scaffolding (`mode=unmanaged`, and unlike the managed variant `network_os` is not required), `interface_type` default `subInterface`, and a `from_config`↔`from_response` round-trip asserting `to_payload` re-emits `interfaceType` / `mode` / `networkOSType` / `policyType`.
- **extra="ignore"** — GET-only fields (e.g. `templateName`/`templateConfig` from the `userDefined` branch) are silently discarded on read.
- Plus composite identifier, `payload_exclude_fields` (drops `switchIp`/`operData`), `operData` alias parsing, `from_response` robustness to missing nesting, and the argspec contract (only `switch_ip` + `interface_name` exposed; all `config_data` hidden).

Modeled on the `nd_interface_subinterface_managed` model tests; constructs inline, so no fixture file was needed.

Verification: `ndpytest` model file **27 passed** (full unmanaged model+orchestrator suite **50 passed**); `black`/`isort` clean; `pylint` clean (only the standard collection-path import-errors common to all test files in this repo).

File: plugins/modules/nd_interface_subinterface_unmanaged.py (hunk 2 above)
Comment (allenrobel) [reply plugins/modules/nd_interface_subinterface_unmanaged.py:67]: Acknowledged — no change in this PR. As noted, `gathered` is tracked collection-wide in #340 for all `NDStateMachine` + orchestrator modules, so `nd_interface_subinterface_unmanaged` will pick it up consistently when that lands (with documented filter semantics, a `gathered` example, and a read-only test asserting no create/update/delete/deploy call is made). Keeping this thread open as the tracking marker for this module.

File: plugins/modules/nd_interface_subinterface_unmanaged.py (hunk 1 above)
Comment (allenrobel) [reply plugins/modules/nd_interface_subinterface_unmanaged.py]: Note: this branch was rebased onto the latest `develop` during the RETURN-block rollout, so the commit referenced above (`ace5964`) is now orphaned. The same change is now `fcd86af9`. Resolving as addressed.

**Review by mikewiebe — APPROVED** [review]

**Review by gmicol — APPROVED** [review]
LGTM
